### PR TITLE
Add merge instructions to releasing docs

### PR DESCRIPTION
**Pull request by jonaf**

### CODE REVIEW

File: docs/Contributing.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Thanks for your interest in contributing! Here are some things you should know.
 To get started:
 
 - fork this project on github
-- create a new branch named after the change you want to make; i.e., `git checkout -b myuser-mynewfeature`
+- create a new branch named after the change you want to make; i.e., `git checkout -b mynewfeature`
```
Comment (temujin9) [inline docs/Contributing.md]: I think I actually prefer 'myuser-mynewfeature'; bulkier, but easier to keep track of.

**Comment by temujin9** [comment]
LGTM otherwise. Let me know when you've tweaked, and I'll try out that fancy new git-land script.

File: docs/Contributing.md (hunk 1 above)
Comment (jonaf) [inline docs/Contributing.md]: When you create a branch named `mynewfeature` on a fork, sending a pull request creates a branch in the upstream repo named `<myuser>-mynewfeature`. I updated the documentation to reflect what actually happens.

Comment (temujin9) [inline docs/Contributing.md]: Well, fine, go ahead and make sense, why don'tcha.

**Pull request merged**